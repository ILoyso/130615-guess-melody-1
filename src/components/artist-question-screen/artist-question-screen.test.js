import React from 'react';
import renderer from 'react-test-renderer';

import ArtistQuestionScreen from './artist-question-screen.jsx';


const questionMock = {
  answers: [
    {
      artist: `Jared Leto`,
      picture: `http://placehold.it/134x134`,
    },
    {
      artist: `Dave Grohl`,
      picture: `http://placehold.it/134x134`,
    },
    {
      artist: `Kurt Cobain`,
      picture: `http://placehold.it/134x134`,
    },
  ],
  onAnswer: jest.fn(),
  question: {
    song: {
      artist: `Kurt Cobain`,
      src: `path.mp3`,
    },
    type: `artist`
  }
};

it(`ArtistQuestionScreen correctly renders`, () => {
  const artistScreen = renderer
    .create(<ArtistQuestionScreen
      question={questionMock.question}
      onAnswer={questionMock.onAnswer}
    />)
    .toJSON();

  expect(artistScreen).toMatchSnapshot();
});
