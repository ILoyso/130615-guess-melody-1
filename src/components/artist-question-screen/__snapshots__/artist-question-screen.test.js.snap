// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen correctly renders 1`] = `
<section
  className="game__screen"
>
  <h2
    className="game__title"
  >
    Кто исполняет эту песню?
  </h2>
  <div
    className="game__track"
  >
    <button
      className="track__button track__button--play"
      disabled={true}
      onClick={[Function]}
      type="button"
    />
    <div
      className="track__status"
    >
      <audio />
    </div>
  </div>
  <form
    className="game__artist"
  >
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="artist-0"
        name="answer"
        onClick={[Function]}
        type="radio"
        value="artist-0"
      />
      <label
        className="artist__name"
        htmlFor="artist-0"
      >
        <img
          alt="Jared Leto"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        Jared Leto
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="artist-1"
        name="answer"
        onClick={[Function]}
        type="radio"
        value="artist-1"
      />
      <label
        className="artist__name"
        htmlFor="artist-1"
      >
        <img
          alt="Dave Grohl"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        Dave Grohl
      </label>
    </div>
    <div
      className="artist"
    >
      <input
        className="artist__input visually-hidden"
        id="artist-2"
        name="answer"
        onClick={[Function]}
        type="radio"
        value="artist-2"
      />
      <label
        className="artist__name"
        htmlFor="artist-2"
      >
        <img
          alt="Kurt Cobain"
          className="artist__picture"
          src="http://placehold.it/134x134"
        />
        Kurt Cobain
      </label>
    </div>
  </form>
</section>
`;
